### PR TITLE
Operation/adding reset to transceiver

### LINKED ISSUE #185: Add Antenna Reset
Expose a simple API from within `RAntenna` that performs a full reset of the antenna system. Something like `int antennaReset()` would be ideal.

Some questions / gotchya's to look out for:
- The underlying SSI function requests `index` and `side` inputs; what do those mean in the context of resetting? Are multiple calls required to fully reset the board (if so, those multiple calls should be made within a single `antennaReset()` call)
- What happens if this `antennaReset()` function is called mid-deployment? Anything, or nothing? Can we prevent / check to make sure that doesn't happen?

### PR DESCRIPTION
**Pull request by ATGr8**

### CODE REVIEW

**Review by TheDopplerEffects — CHANGES_REQUESTED** [review]
The RAntenna files bled into this PR. You will have to fix that.

**Comment by ATGr8** [comment]
fixed!

File: radsat-sk/operation/subsystems/RAntenna.h [hunk 1]
```diff
@@ -59,6 +59,6 @@ typedef struct _antenna_telemetry_t {
 int antennaInit(void);
 int antennaDeploymentAttempt(void);
 int antennaTelemetry(antenna_telemetry_t* telemetry);
-
+int antennaReset(void);
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RAntenna.h]: Shouldn't be included in this PR

File: radsat-sk/operation/subsystems/RTransceiver.c [hunk 2]
```diff
@@ -2,6 +2,8 @@
  * @file RTransceiver.h
  * @date December 23, 2021
  * @author Tyrel Kostyk (tck290) and Matthew Buglass (mab839)
+ * @edit May 8,2022
+ * @edit author Atharva(iya789)
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RTransceiver.c]: See comments on Antenna PR. Edit flag not necessary

File: radsat-sk/operation/subsystems/RTransceiver.c [hunk 3]
```diff
@@ -281,10 +283,25 @@ int transceiverResetWatchDogs(void) {
...
+
+	int error = IsisTrxvu_softReset(TRANSCEIVER_RX_I2C_SLAVE_ADDR);
+
+	//TODO: record errors (if present) to System Manager
+	if (error != 0)
+			return error;
+
+	return 0;
+}
+
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RTransceiver.c]: - White space and indentation issues; follow example of other functions in this file
- Instead of providing a slave addr, I believe this `IsisTrxvu_softReset` is asking an _index_; see the `transceiverPowerCycle` function above for an example
- You should "move" this function to be next to the `transceiverPowerCycle` function in this file, just so they're grouped together since they have very similar functionality

File: radsat-sk/operation/subsystems/RTransceiver.c [hunk 4]
```diff
@@ -281,10 +283,25 @@ int transceiverResetWatchDogs(void) {
...
 
 /***************************************************************************************************
                                          PRIVATE FUNCTIONS
-***************************************************************************************************/
+*******************************************  ********************************************************/
```
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RTransceiver.c]: Remove space lol

**Review by tyrelkostyk — DISMISSED** [review]
Handful of style and functionality changes requested.

**Comment by tyrelkostyk** [comment]
You have re-requested review, however you have not addressed all of the comments from the previous review. I believe the transceiver SSI functions are requiring the `TRANSCEIVER_INDEX` macro as input arguments, not the slave address itself.

**Comment by ATGr8** [comment]
My bad! I used the transceiver function this time

File: radsat-sk/operation/subsystems/RTransceiver.c (hunk 4 above)
Comment (TheDopplerEffects) [reply radsat-sk/operation/subsystems/RTransceiver.c]: Didn't do

**Comment by TheDopplerEffects** [comment]
Please be more methodical when going over our comments. All I want to do is merge this PR.

**Review by TheDopplerEffects — APPROVED** [review]
GTG
